File: web/src/app/models/batch.ts
import { z } from 'zod';
import { contactSchema } from './contact';
import { statusSchema } from './status';
import { templateSchema } from './templates';

export const batchContactSchema = contactSchema.merge(
  z.object({
    hidden: z.boolean(),
    status: statusSchema,
  }),
);

export const batchSchema = z.object({
  id: z.string(),
  state: statusSchema,
  name: z.string(),
  contacts: z.array(batchContactSchema),
  createdAt: z.coerce.date(),
  lastModified: z.coerce.date(),
  templates: z.array(templateSchema),
});

export type Batch = z.infer<typeof batchSchema>;

File: comsat/src/main/java/nl/codefusion/comsat/models/batch.java
package nl.codefusion.comsat.models;

public class batch {
    private String id;
    private String name;
    private String state;
    private String lastModified;
    private String createdAt;

    public void setLastModified(String format) {
        this.lastModified = format;
    }

    public void setState(String processed) {
        this.state = processed;
    }

}
